### PR TITLE
Enable duplication engine automatically for supported languages

### PR DESCRIPTION
**Pull request by dblandin**
Duplicate of #259 

This PR updates our engines registry to enable the [duplication engine][] for any project with Ruby, Python, JavaScript, or PHP files.

[duplication engine]: https://circleci.com/gh/codeclimate/codeclimate-duplication

### CODE REVIEW

File: config/engines.yml [hunk 1]
```diff
@@ -55,7 +55,17 @@ duplication:
   description: Structural duplication detection for Ruby, Python, JavaScript, and PHP.
   community: false
   enable_regexps:
+    - \.js$
+    - \.jsx$
+    - \.php$
```
Comment (GordonDiggs) [inline config/engines.yml]: PHP [has a few extensions](https://github.com/codeclimate/codeclimate-duplication/blob/master/lib/cc/engine/analyzers/php/main.rb#L13-L15)

File: spec/cc/cli/init_spec.rb [hunk 2]
```diff
@@ -170,11 +171,12 @@ module CC::CLI
           YAML.safe_load(new_content).must_equal({
             "engines" => {
               "csslint" => { "enabled"=>true },
+              "duplication" => { "enabled" => true, "config" => { "languages" => ["ruby", "javascript", "python", "php"] } },
```
Comment (GordonDiggs) [inline spec/cc/cli/init_spec.rb]: Is this array format still valid? I've been doing them as mappings recently

Comment (dblandin) [inline spec/cc/cli/init_spec.rb]: Looks like the array format is still [valid](https://github.com/codeclimate/codeclimate-duplication/blob/ce27d9e3b99513115968df64eafa13c8693d5a3c/lib/cc/engine/analyzers/engine_config.rb#L57-L67). The hash format is necessary when you're overriding mass threshold for a language.

File: config/engines.yml (hunk 1 above)
Comment (dblandin) [inline config/engines.yml]: @GordonDiggs Are those extensions common enough to include by default or should we leave it as an opt in? If we add those two other extensions here, we should probably add them for ratings as well.

Comment (GordonDiggs) [inline config/engines.yml]: They are definitely common. I would support adding them in both

Comment (dblandin) [inline config/engines.yml]: Sounds good :+1:

**Comment by dblandin** [comment]
Updated to include on `.inc` and `.module` files.

File: spec/cc/cli/init_spec.rb (hunk 2 above)
Comment (GordonDiggs) [inline spec/cc/cli/init_spec.rb]: good to know! thanks

**Comment by GordonDiggs** [comment]
:+1:

**Pull request merged**